x-pack/plugins/features/server/feature_registry.ts

```diff
@@ -169,19 +190,106 @@ export class FeatureRegistry {
     }
   }
 
-  public getAllKibanaFeatures(license?: ILicense, ignoreLicense = false): KibanaFeature[] {
+  /**
+   * Once all features are registered and the registry is locked, this method should validate the integrity of the registered feature set, including any potential cross-feature dependencies.
+   */
+  public validateFeatures() {
     if (!this.locked) {
-      throw new Error('Cannot retrieve Kibana features while registration is still open');
+      throw new Error(
+        'Cannot validate features while the registry is not locked and still allows further feature registrations.'
+      );
+    }
+
+    for (const feature of Object.values(this.kibanaFeatures)) {
+      if (!feature.privileges) {
+        continue;
+      }
+
+      // Iterate over all top-level and sub-feature privileges.
+      const isFeatureDeprecated = !!feature.deprecated;
+      for (const [privilegeId, privilege] of [
```

Comment:
__note:__ this is not an exhaustive list of early checks and guardrails we can implement, but it certainly covers 90%+ of the errors we expect developers might make initially.